### ISSUE TITLE
Out of memory

### ISSUE THREAD

**Issue by cntontis** [joblib/joblib#1129]
I am using joblib to download in parallel some json files.
There are ~100.000 json files (~30kB/file). I have a 16 core machine with 64GB of memory, which i suppose should be enough for this job since the total size of the jsons is 3GB. the code bellow is a minimal reproduction of what I am trying to do. Also its worth mentioning that the code is running inside a docker container. Though I tried running it without it and the same thing happens the memory keeps raising and eventually it errors with an out of memory indication.
```python

def read_json_from_s3(path: str):
  json_file = download_json_from_s3(path)
  json_dict = json.load(json_file)

jsons = Parallel(n_jobs=n_cores)(
            delayed(read_json_from_s3)(path_to_s3) for path_to_s3 in paths_to_s3)
```

**Comment by manojbalaji1** [comment]
I have come across a similar issue, I think memory is not being released, can you check if it is the same for you @cntontis ?

**Comment by cntontis** [comment]
Memory was being released fine.
it was my mistake, the actual size of the json was bigger than expected and that's why it was running out of memory.

**Comment by tomMoral** [comment]
Thanks for the report @cntontis 

@manojbalaji1 the memory is supposed to be released by `joblib` and if you install `psutil`, we also have some checks that kill workers where the memory increased a lot after a task. If you see memory not being released by `joblib`, can you open a new issue with a reproducing example? Thanks.
